### PR TITLE
fix #1876

### LINKED ISSUE #1876: the default value of setAllowFileAccess was changed since Android 11(MSTG-PLATFORM-6)
**Describe the issue**
Since Android 11(targetSDKVersion 30), the default value of setAllowFileAccess has changed from true to false.
- [developer.android.com: setAllowFileAccess](https://developer.android.com/reference/android/webkit/WebSettings#setAllowFileAccess(boolean))

so I think it is better to make the following modifications.

```diff
-`setAllowFileAccess`: Enables and disables file access within a WebView. File access is enabled by default. Note that this enables and disables [file system access](https://developer.android.com/reference/android/webkit/WebSettings.html#setAllowFileAccess%28boolean%29 "File Access in WebView") only. Asset and resource access is unaffected and accessible via `file:///android_asset` and `file:///android_res`.
+`setAllowFileAccess`: Enables and disables file access within a WebView. The default value is `true` for Android 10 (API level 29) and below and `false` for Android 11 (API level 30) and above. Note that this enables and disables [file system access](https://developer.android.com/reference/android/webkit/WebSettings.html#setAllowFileAccess%28boolean%29 "File Access in WebView") only. Asset and resource access is unaffected and accessible via `file:///android_asset` and `file:///android_res`.
```

**Optional: Testcase or section**
[MSTG-PLATFORM-6](https://github.com/OWASP/owasp-mstg/blob/master/Document/0x05h-Testing-Platform-Interaction.md)

**Optional: Additional context**
If there is no other opinions, I will send a PR later.

### PR DESCRIPTION
**Pull request by fujiokayu**
Thank you for submitting a Pull Request to the Mobile Security Testing Guide. Please make sure that:

- [ ] Your contribution is written in the 2nd person (e.g. you)
- [x] Your contribution is written in an active present form for as much as possible.
- [x] You have made sure that the reference section is up to date (e.g. please add sources you have used, make sure that the references to MITRE/MASVS/etc. are up to date)
- [x] Your contribution has proper formatted markdown and/or code
- [ ] Any references to website have been formatted as [TEXT](URL “NAME”)
- [ ] You verified/tested the effectiveness of your contribution (e.g.: is the code really an effective remediation? Please verify it works!)

If your PR is related to an issue. Please end your PR test with the following line:
This PR closes #1876 .

---

- references
  - [developer.android.com: setAllowFileAccess](https://developer.android.com/reference/android/webkit/WebSettings#setAllowFileAccess(boolean))

### CODE REVIEW

File: Document/0x05h-Testing-Platform-Interaction.md [hunk 1]
```diff
@@ -1130,7 +1130,7 @@ WebViews can load remote content from an endpoint, but they can also load local
...
 
 - `setAllowContentAccess`: Content URL access allows WebViews to load content from a content provider installed on the system, which is enabled by default .
-- `setAllowFileAccess`: Enables and disables file access within a WebView. File access is enabled by default. Note that this enables and disables [file system access](https://developer.android.com/reference/android/webkit/WebSettings.html#setAllowFileAccess%28boolean%29 "File Access in WebView") only. Asset and resource access is unaffected and accessible via `file:///android_asset` and `file:///android_res`.
+- `setAllowFileAccess`: Enables and disables file access within a WebView. The default value is `true` for Android 10 (API level 29) and below and `false` for Android 11 (API level 30) and above. Note that this enables and disables [file system access](https://developer.android.com/reference/android/webkit/WebSettings.html#setAllowFileAccess%28boolean%29 "File Access in WebView") only. Asset and resource access is unaffected and accessible via `file:///android_asset` and `file:///android_res`.
```
Comment (cpholguera) [inline Document/0x05h-Testing-Platform-Interaction.md:1133]: What do you think of this? The app has to actively target the version, right? Probably it doesn't suffice to be run on a system having the newest version.


```suggestion
- `setAllowFileAccess`: Enables and disables file access within a WebView. The default value is `true` when targeting Android 10 (API level 29) and below and `false` for Android 11 (API level 30) and above. Note that this enables and disables [file system access](https://developer.android.com/reference/android/webkit/WebSettings.html#setAllowFileAccess%28boolean%29 "File Access in WebView") only. Asset and resource access is unaffected and accessible via `file:///android_asset` and `file:///android_res`.
```

**Review by cpholguera — CHANGES_REQUESTED** [review]
Very nice update. Just one little suggestion :)

File: Document/0x05h-Testing-Platform-Interaction.md (hunk 1 above)
Comment (fujiokayu) [reply Document/0x05h-Testing-Platform-Interaction.md]: Thanks, I totally agree with your suggestion.
Fixed in commit e9d5763f0de536598429b9947c6dc09d4d0c2211

**Review by cpholguera — APPROVED** [review]
Thank you for this update @fujiokayu, keep them gladly coming! 😊

**Pull request merged**